anchors/dna/anchors/anchors.js

```
@@ -1,528 +1,99 @@
-
-    catch (err) // getLink got an error
-    {
-        if (err.holochainMessage == "hash not found") // hash not found, create the base entry, and continue with empty links list
-        {
-            var anchorHash = commit("anchor_base", anchor);
-            links = [];
-            newAnchor = true;
-        }
-        else if (err.holochainMessage == "No links for " + ANCHOR_GENERIC)
-            throw new errorObject("\"" + anchor + "\" is not a simple anchor link base!");
-        else
-            throw err; // other error, throw it
-
-    }
-
-    if (links.length != 1 && !newAnchor) // an existing Key/Value base entry will only always have just 1 link entry, no more, no less
-        throw new errorObject("\"" + anchor + "\" is not a simple anchor link base!");
-
-    // validate value parameters and generate valueHash if needed
-    if (valueHash == undefined) {
-        if (value == undefined)
-            throw new errorObject("Must pass either value or valueHash!");
-
-        valueHash = commit(entryType, makeAnchorValue(entryType, value));
-    }
-    else if (value != undefined)
-        throw new errorObject("Can't pass both value and valueHash!");
-
-    // if this is not a new key, delete the old value and link
-    if (links.length == 1) {
-
-        commit("anchor_link", { Links: [{ Base: anchorHash, Link: links[0].Hash, Tag: ANCHOR_GENERIC, LinkAction: HC.LinkAction.Del }] });
-
-        if (!preserveOldValueEntry)
-            remove(links[0].Hash, "anchors-set");
-
-    }
-
-    // add the new value's link
-    var link_hash = commit("anchor_link", { Links: [{ Base: anchorHash, Link: valueHash, Tag: ANCHOR_GENERIC }] });
-
-    return { anchorHash: anchorHash, valueHash: valueHash, linkHash: link_hash };
-
+/**
+ * @param type (string)
+ * @return array of anchor entries
+ **/
+function anchors(type){
+  var links = getLinks(makeHash('anchor', {anchorType: type, anchorText: ''}), '');
+  return links;
```

Should add some error checking in case links is not an array, or return an empty array to deal with the absence gracefully.

---

In this one it looks like errors got set to ThrowErrors instead of return, so if it doesn't have any it won't return!

---

Does that mean it will just crash? Would we do a try/catch here?  Or we add to @return that it will return an array or the consuming function should catch the error.